### DISCUSSION TITLE
Hyperlink modification

### DISCUSSION THREAD

**Discussion by MathisICDC** [opendcim/openDCIM#1470] [Q&A] [answered]
Hello everyone,

I'm experiencing a small problem for my users who only have read access to openDCIM. 

When they search for a device, they can't click on it directly. They have to go through the cabinet and then click on the device again (there's no hyperlink). But when I activate the right to modify assets, the hyperlink is present when they do a search. 
What can I do to enable the hyperlink to be read-only so that they can click directly on the device when doing a search?

Without hyperlink
![without hyperlink](https://github.com/opendcim/openDCIM/assets/148790073/7c3688e3-e41f-4569-ac7c-5d408c0374fe)

With hyperlink
![with hyperlink](https://github.com/opendcim/openDCIM/assets/148790073/60306a6c-2b43-482f-bfcf-b48482e3c1c8)

Thank you in advance.

Best regards,
Mathis

**Comment by wilpig** [comment]
I'll look into it. It's likely in how we check for rights when we build out the search results.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: MathisICDC ***@***.***>
Sent: Monday, October 23, 2023 10:29:23 AM
To: opendcim/openDCIM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [opendcim/openDCIM] Hyperlink modification (Discussion #1470)


Hello everyone,

I'm experiencing a small problem for my users who only have read access to openDCIM.

When they search for a device, they can't click on it directly. They have to go through the cabinet and then click on the device again (there's no hyperlink). But when I activate the right to modify assets, the hyperlink is present when they do a search.
What can I do to enable the hyperlink to be read-only so that they can click directly on the device when doing a search?

Without hyperlink
[without hyperlink]<https://user-images.githubusercontent.com/148790073/277390565-7c3688e3-e41f-4569-ac7c-5d408c0374fe.PNG>

With hyperlink
[with hyperlink]<https://user-images.githubusercontent.com/148790073/277390581-60306a6c-2b43-482f-bfcf-b48482e3c1c8.PNG>

Thank you in advance.

Best regards,
Mathis

—
Reply to this email directly, view it on GitHub<https://github.com/opendcim/openDCIM/discussions/1470>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAMTY6RWVT45K2UAFCPD5ATYA2ENHAVCNFSM6AAAAAA6MISANOVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVG43DQNRRHA>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/opendcim/openDCIM/discussions/1470", "url": "https://github.com/opendcim/openDCIM/discussions/1470", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by MathisICDC** [reply]
Hello @wilpig,
Have you had time to look at this?
Thank you in advance

**Reply by wilpig** [reply]
no

**Reply by wilpig** [reply]
Pretty sure that I just need to change this from ==Write to !=None

https://github.com/opendcim/openDCIM/blob/master/search.php#L317

I need to do some testing to make sure i'm not forgetting anything though.

**Reply by wilpig** [reply, marked as answer]
https://github.com/opendcim/openDCIM/commit/036e6798a31fc9e12e82813030acaf7b176b0f7a
